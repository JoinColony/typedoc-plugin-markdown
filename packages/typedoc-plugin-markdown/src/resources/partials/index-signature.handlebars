\[{{#each parameters}}{{name}}: {{{type this.type}}}{{/each}}\]: {{{type type}}}

{{> comment}}