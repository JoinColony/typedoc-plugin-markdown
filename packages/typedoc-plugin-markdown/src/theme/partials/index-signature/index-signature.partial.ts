import { SignatureReflection } from 'typedoc';
import { MarkdownThemeRenderContext } from '../../theme.context';
import { backTicks } from '../../theme.utils';

export const indexSignaturePartial = (
  context: MarkdownThemeRenderContext,
  props: SignatureReflection,
) => {
  const parameters = props.parameters
    ? props.parameters.map((parameter) => {
        return `${backTicks(parameter.name)}: ${context.typePartial(
          parameter.type,
        )}`;
      })
    : [];
  return `\\[${parameters.join('')}\\]: ${context.typePartial(props.type)}`;
};
