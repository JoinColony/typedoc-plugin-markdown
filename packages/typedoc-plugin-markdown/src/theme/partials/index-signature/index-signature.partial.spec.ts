import { ProjectReflection } from 'typedoc';
import { indexSignaturePartial } from './index-signature.partial';

describe(`Index Signature Partial:`, () => {
  let project: ProjectReflection;

  beforeAll(async () => {
    project = global.getProject(__dirname);
  });

  test(`should compile index signature`, () => {
    expect(
      indexSignaturePartial(
        global.getMockContext(),
        (project.findReflectionByName('indexableDeclaration') as any).type
          .declaration.indexSignature,
      ),
    ).toMatchSnapshot();
  });
});
