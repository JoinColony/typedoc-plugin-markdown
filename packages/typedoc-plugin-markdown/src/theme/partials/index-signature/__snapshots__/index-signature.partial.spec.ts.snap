// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Signature Partial: should compile index signature 1`] = `"\\\\[\`index\`: { typePartial }\\\\]: { typePartial }"`;
